src/components/common/ContentSwitch.tsx

```diff
@@ -0,0 +1,150 @@
+"use client";
+
+import { Resources } from "contentlayer/generated";
+import React, { SetStateAction, useState } from "react";
+import TranscriptTabContent from "../individual-transcript/TranscriptTabContent";
+import ContentGrouping from "../explore/ContentGrouping";
+import { TopicsData } from "@/utils";
+
+const ContentSwitch = ({
+  summary,
+  markdown,
+  extraInfo,
+  currentHeading,
+  groupedHeading,
+  setCurrentHeading
+}: {
+  summary?: string;
+  markdown: string;
+  extraInfo?: Resources[];
+  currentHeading?: string;
+  groupedHeading?: Record<string, TopicsData[]>;
+  setCurrentHeading?: React.Dispatch<SetStateAction<string>>
+}) => {
+  const [openSection, setOpenSection] = useState({
+    transcript: true,
+    summary: false,
+    extraInfo: false,
+  });
+
```

Comment:
You are using different names to refer to the same things. Here you call these tabs "sections", but in the component level you call them "switch items".

Additionally, on the implementation, when you have only one item selected, it's common to use something like this
```suggestion
  const [openSection, setOpenSection] = useState<"transcript" | "summary" | "extraInfo">("transcript")
```

cause you don't expect that more than one of those values will need to be "true", so it simplifies the code.